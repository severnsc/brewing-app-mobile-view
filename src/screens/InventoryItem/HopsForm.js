import React from "react";
import PropTypes from "prop-types";
import { Icon, TextInput } from "../../components";

const HopsForm = ({
  currentQuantity,
  setCurrentQuantity,
  alphaAcids,
  setAlphaAcids,
  country,
  setCountry,
  unitCost,
  setUnitCost,
  deliveryDate,
  setDelvieryDate,
  reorderQuantity,
  setReorderQuantity,
  reorderThreshold,
  setReorderThreshold
}) => (
  <React.Fragment>
    <Icon name="balance-scale" />
    <TextInput
      placeholder="Current Quantity"
      value={currentQuantity}
      onChange={setCurrentQuantity}
      keyboardType="decimal-pad"
    />
    <Icon name="percent" />
    <TextInput
      placeholder="Alpha Acids"
      value={alphaAcids}
      onChange={setAlphaAcids}
      keyboardType="decimal-pad"
    />
    <Icon name="md-globe" />
    <TextInput
      placeholder="Country of Origin"
      value={country}
      onChange={setCountry}
    />
    <Icon name="ios-cash" />
    <TextInput
      placeholder="Unit Cost"
      value={unitCost}
      onChange={setUnitCost}
      keyboardType="decimal-pad"
    />
    <Icon name="ios-calendar" />
    <TextInput
      placeholder="Delivery Date"
      value={deliveryDate}
      onChange={setDelvieryDate}
    />
    <Icon name="ios-list" />
    <TextInput
      placeholder="Amount to Reorder"
      value={reorderQuantity}
      onChange={setReorderQuantity}
      keyboardType="decimal-pad"
    />
    <Icon name="ios-list" />
    <TextInput
      placeholder="Quantity at which to Reorder"
      value={reorderThreshold}
      onChange={setReorderThreshold}
      keyboardType="decimal-pad"
    />
  </React.Fragment>
);

HopsForm.propTypes = {
  currentQuantity: PropTypes.number,
  setCurrentQuantity: PropTypes.func.isRequired,
  alphaAcids: PropTypes.number,
  setAlphaAcids: PropTypes.func.isRequired,
  country: PropTypes.string,
  setCountry: PropTypes.func.isRequired,
  unitCost: PropTypes.number,
  setUnitCost: PropTypes.func.isRequired,
  deliveryDate: PropTypes.string,
  setDelvieryDate: PropTypes.func.isRequired,
  reorderQuantity: PropTypes.number,
  setReorderQuantity: PropTypes.func.isRequired,
  reorderThreshold: PropTypes.number,
  setReorderThreshold: PropTypes.func.isRequired
};

HopsForm.defaultProps = {
  setAlphaAcids: () => {},
  setCountry: () => {},
  setCurrentQuantity: () => {},
  setDelvieryDate: () => {},
  setReorderQuantity: () => {},
  setReorderThreshold: () => {},
  setUnitCost: () => {}
};

export default HopsForm;
