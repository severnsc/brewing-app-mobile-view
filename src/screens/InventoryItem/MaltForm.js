import React from "react";
import PropTypes from "prop-types";
import { Text, TextInput, Icon, MaltColor } from "../../components";

const MaltForm = ({
  currentQuantity,
  setCurrentQuantity,
  type,
  selectMaltType,
  color,
  setColor,
  country,
  setCountry,
  unitCost,
  setUnitCost,
  deliveryDate,
  setDeliveryDate,
  reorderQuantity,
  setReorderQuantity,
  reorderThreshold,
  setReorderThreshold
}) => (
  <React.Fragment>
    <Icon name="balance-scale" />
    <TextInput
      placeholder="Current Quantity"
      value={currentQuantity}
      onChange={setCurrentQuantity}
    />
    <Icon name="ios-list" />
    <Text value={type} onPress={selectMaltType} />
    <Icon name="ios-arrow-forward" />
    <MaltColor value={2} displayValue={false} />
    <TextInput placeholder="Malt Color" value={color} onChange={setColor} />
    <Icon name="md-globe" />
    <TextInput
      placeholder="Country of Origin"
      value={country}
      onChange={setCountry}
    />
    <Icon name="ios-cash" />
    <TextInput
      placeholder="Unit Cost"
      value={unitCost}
      onChange={setUnitCost}
    />
    <Icon name="ios-calendar" />
    <TextInput
      placeholder="Delivery Date"
      value={deliveryDate}
      onChange={setDeliveryDate}
    />
    <Icon name="ios-list" />
    <TextInput
      placeholder="Amount to Reorder"
      value={reorderQuantity}
      onChange={setReorderQuantity}
    />
    <Icon name="ios-list" />
    <TextInput
      placeholder="Quantity at which to Reorder"
      value={reorderThreshold}
      onChange={setReorderThreshold}
    />
    );
  </React.Fragment>
);

MaltForm.propTypes = {
  currentQuantity: PropTypes.number,
  setCurrentQuantity: PropTypes.func.isRequired,
  type: PropTypes.oneOf(["Base", "Crystal", "Roast", "Malt Type"]),
  selectMaltType: PropTypes.func.isRequired,
  color: PropTypes.number,
  setColor: PropTypes.func.isRequired,
  country: PropTypes.string,
  setCountry: PropTypes.func.isRequired,
  unitCost: PropTypes.number,
  setUnitCost: PropTypes.func.isRequired,
  deliveryDate: PropTypes.string,
  setDeliveryDate: PropTypes.func.isRequired,
  reorderQuantity: PropTypes.number,
  setReorderQuantity: PropTypes.func.isRequired,
  reorderThreshold: PropTypes.number,
  setReorderThreshold: PropTypes.func.isRequired
};

MaltForm.defaultProps = {
  setColor: () => {},
  setCountry: () => {},
  setCurrentQuantity: () => {},
  setDeliveryDate: () => {},
  setReorderQuantity: () => {},
  setReorderThreshold: () => {},
  selectMaltType: () => {},
  setUnitCost: () => {}
};

export default MaltForm;
