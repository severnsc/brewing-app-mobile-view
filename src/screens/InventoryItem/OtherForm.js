import React from "react";
import { Icon, TextInput } from "../../components";

const OtherForm = () => (
  <React.Fragment>
    <Icon name="balance-scale" />
    <TextInput placeholder="Current Quantity" />
    <Icon name="ios-cash" />
    <TextInput placeholder="Unit Cost" />
    <Icon name="ios-calendar" />
    <TextInput placeholder="Delivery Date" />
    <Icon name="ios-list" />
    <TextInput placeholder="Amount to Reorder" />
    <Icon name="ios-list" />
    <TextInput placeholder="Quantity at which to Reorder" />
  </React.Fragment>
);

export default OtherForm;
