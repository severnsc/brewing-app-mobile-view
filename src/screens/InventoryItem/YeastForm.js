import React from "react";
import PropTypes from "prop-types";
import { TextInput, Icon, Text } from "../../components";
import { Switch } from "react-native";

const YeastForm = ({
  currentQuantity,
  setCurrentQuantity,
  type,
  selectType,
  isLiquid,
  toggleIsLiquid,
  unitCost,
  setUnitCost,
  deliveryDate,
  setDeliveryDate,
  reorderQuantity,
  setReorderQuantity,
  reoderThreshold,
  setReorderThreshold
}) => (
  <React.Fragment>
    <Icon name="balance-scale" />
    <TextInput
      placeholder="Current Quantity"
      value={currentQuantity}
      onChange={setCurrentQuantity}
    />
    <Icon name="ios-list" />
    <Text value="Yeast Type" value={type} onPress={selectType} />
    <Icon name="ios-arrow-forward" />
    <Icon name="ios-flask" />
    <Text value="Dry" />
    <Switch value={isLiquid} onValueChange={toggleIsLiquid} />
    <Text value="Liquid" />
    <Icon name="md-globe" />
    <Icon name="ios-cash" />
    <TextInput
      placeholder="Unit Cost"
      value={unitCost}
      onChange={setUnitCost}
    />
    <Icon name="ios-calendar" />
    <TextInput
      placeholder="Delivery Date"
      value={deliveryDate}
      onChange={setDeliveryDate}
    />
    <Icon name="ios-list" />
    <TextInput
      placeholder="Amount to Reorder"
      value={reorderQuantity}
      onChange={setReorderQuantity}
    />
    <Icon name="ios-list" />
    <TextInput
      placeholder="Quantity at which to Reorder"
      value={reoderThreshold}
      onChange={setReorderThreshold}
    />
  </React.Fragment>
);

YeastForm.propTypes = {
  currentQuantity: PropTypes.number,
  setCurrentQuantity: PropTypes.func.isRequired,
  type: PropTypes.oneOf(["Yeast Type", "Ale", "Lager", "Hybrid"]),
  selectType: PropTypes.func.isRequired,
  isLiquid: PropTypes.bool,
  toggleIsLiquid: PropTypes.func.isRequired,
  unitCost: PropTypes.number,
  setUnitCost: PropTypes.func.isRequired,
  deliveryDate: PropTypes.string,
  setDeliveryDate: PropTypes.func.isRequired,
  reorderQuantity: PropTypes.number,
  setReorderQuantity: PropTypes.func.isRequired,
  reoderThreshold: PropTypes.number,
  setReorderThreshold: PropTypes.func.isRequired
};

YeastForm.defaultProps = {
  type: "Yeast Type",
  setCurrentQuantity: () => {},
  selectType: () => {},
  toggleIsLiquid: () => {},
  setUnitCost: () => {},
  setDeliveryDate: () => {},
  setReorderQuantity: () => {},
  setReorderThreshold: () => {}
};

export default YeastForm;
