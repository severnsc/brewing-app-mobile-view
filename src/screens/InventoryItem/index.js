import React, { useState } from "react";
import PropTypes from "prop-types";
import HopsForm from "./HopsForm";
import MaltForm from "./MaltForm";
import YeastForm from "./YeastForm";
import OtherForm from "./OtherForm";
import { TextInput, Icon, Text } from "../../components";
import { StyleSheet, View } from "react-native";

const styles = StyleSheet.create({
  title: {
    fontSize: 40
  }
});

const ItemName = () => {
  const [value, setValue] = useState("");
  return (
    <TextInput
      value={value}
      onChange={setValue}
      style={styles.title}
      placeholder="Item Name"
      testID="itemName"
    />
  );
};

const PurchaseDate = () => {
  const [value, setValue] = useState("");
  return (
    <TextInput
      value={value}
      onChange={setValue}
      placeholder="Purchase Date"
      testID="purchaseDate"
    />
  );
};

const renderForm = selected => {
  switch (selected) {
    case "Malt":
      return <MaltForm />;
    case "Yeast":
      return <YeastForm />;
    case "Other":
      return <OtherForm />;
    default:
      return <HopsForm />;
  }
};

const InventoryItem = ({ navigation: { navigate }, setSelected, selected }) => {
  const onPress = () => navigate("INVENTORY_SELECT", { setSelected });
  return (
    <View>
      <ItemName />
      <PurchaseDate />
      <Text value={selected} onPress={onPress} />
      <Icon name="ios-arrow-forward" />
      {renderForm(selected)}
    </View>
  );
};

InventoryItem.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired,
  setSelected: PropTypes.func.isRequired,
  selected: PropTypes.string.isRequired
};

InventoryItem.defaultProps = {
  navigation: {
    navigate: () => {}
  },
  setSelected: () => {},
  selected: "Hops"
};

export default InventoryItem;
