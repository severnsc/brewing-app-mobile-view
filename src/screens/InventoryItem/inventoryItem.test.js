import React from "react";
import InventoryItem from ".";
import HopsForm from "./HopsForm";
import MaltForm from "./MaltForm";
import YeastForm from "./YeastForm";
import OtherForm from "./OtherForm";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("HopsForm", () => {
  it("matches snapshot when no values are set", () => {
    const hopsForm = renderer.create(<HopsForm />);
    expect(hopsForm).toMatchSnapshot();
  });
});

describe("MaltForm", () => {
  it("matches snapshot when no values are set", () => {
    const maltForm = renderer.create(<MaltForm />);
    expect(maltForm).toMatchSnapshot();
  });
  it("matches snapshot when values are set", () => {
    const maltForm = renderer.create(
      <MaltForm
        currentQuantity={20}
        color={5}
        country="UK"
        unitCost={3}
        deliveryDate="01/01/2019"
        reorderQuantity={30}
        reorderThreshold={10}
      />
    );
    expect(maltForm).toMatchSnapshot();
  });
});

describe("YeastForm", () => {
  it("matches snapshot when no values are set", () => {
    const yeastForm = renderer.create(<YeastForm />);
    expect(yeastForm).toMatchSnapshot();
  });
  it("matches snapshot when values are set", () => {
    const yeastForm = renderer.create(
      <YeastForm
        type="Ale"
        currentQuantity={2}
        setCurrentQuantity={jest.fn()}
        selectType={jest.fn()}
        isLiquid={true}
        toggleIsLiquid={jest.fn()}
        unitCost={10}
        setUnitCost={jest.fn()}
        deliveryDate={"01/02/2019"}
        setDeliveryDate={jest.fn()}
        reorderQuantity={1}
        setReorderQuantity={jest.fn()}
        reoderThreshold={1}
        setReorderThreshold={jest.fn()}
      />
    );
    expect(yeastForm).toMatchSnapshot();
  });
});

describe("otherForm", () => {
  it("matches snapshot", () => {
    const otherForm = renderer.create(<OtherForm />);
    expect(otherForm).toMatchSnapshot();
  });
});

describe("InventoryItem screen", () => {
  it("calls navigate prop with correct args when Inventory Type is pressed", () => {
    const setSelected = jest.fn();
    const navigation = {
      navigate: jest.fn()
    };
    const inventoryItem = shallow(
      <InventoryItem navigation={navigation} setSelected={setSelected} />
    );
    const inventoryType = inventoryItem.find("Text");
    inventoryType.props().onPress();
    expect(navigation.navigate).toHaveBeenCalledWith("INVENTORY_SELECT", {
      setSelected
    });
  });
  it("updates the value of Item Name text input onChange", () => {
    const inventoryItem = shallow(<InventoryItem />);
    const itemName = inventoryItem.find("ItemName");
    const textInput = itemName.dive();
    const newValue = "new value";
    textInput.props().onChange(newValue);
    expect(textInput.prop("value")).toBe(newValue);
  });
  it("updates the value of Purchase Date input onChange", () => {
    const inventoryItem = shallow(<InventoryItem />);
    const purchaseDate = inventoryItem.find("PurchaseDate");
    const input = purchaseDate.dive();
    const newValue = "01/01/2019";
    input.props().onChange(newValue);
    expect(input.prop("value")).toBe(newValue);
  });
  it("returns a hops form by default", () => {
    const inventoryItem = shallow(<InventoryItem />);
    const hopsForm = inventoryItem.find("HopsForm");
    expect(hopsForm).toHaveLength(1);
  });
  it("returns a MaltForm when Malt is selected", () => {
    const inventoryItem = shallow(<InventoryItem selected="Malt" />);
    const maltForm = inventoryItem.find("MaltForm");
    expect(maltForm).toHaveLength(1);
    expect(maltForm).toMatchSnapshot();
  });
  it("returns a YeastForm when Yeast is selected", () => {
    const inventoryItem = shallow(<InventoryItem selected="Yeast" />);
    const yeastForm = inventoryItem.find("YeastForm");
    expect(yeastForm).toHaveLength(1);
    expect(yeastForm).toMatchSnapshot();
  });
  it("returns a OtherForm when Other is selected", () => {
    const inventoryItem = shallow(<InventoryItem selected="Other" />);
    const otherForm = inventoryItem.find("OtherForm");
    expect(otherForm).toHaveLength(1);
    expect(otherForm).toMatchSnapshot();
  });
});
