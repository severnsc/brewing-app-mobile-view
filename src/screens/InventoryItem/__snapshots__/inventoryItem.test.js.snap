// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HopsForm matches snapshot when no values are set 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Current Quantity"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Alpha Acids"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Country of Origin"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Unit Cost"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Delivery Date"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Amount to Reorder"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Quantity at which to Reorder"
    />
  </View>,
]
`;

exports[`HopsForm matches snapshot when values are set 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Current Quantity"
      value="10"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Alpha Acids"
      value="12.4"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Country of Origin"
      value="USA"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Unit Cost"
      value="12"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Delivery Date"
      value="01/01/2019"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Amount to Reorder"
      value="5"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="decimal-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Quantity at which to Reorder"
      value="2"
    />
  </View>,
]
`;

exports[`InventoryItem screen returns a MaltForm when Malt is selected 1`] = `ShallowWrapper {}`;

exports[`InventoryItem screen returns a OtherForm when Other is selected 1`] = `ShallowWrapper {}`;

exports[`InventoryItem screen returns a YeastForm when Yeast is selected 1`] = `ShallowWrapper {}`;

exports[`MaltForm matches snapshot when no values are set 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Current Quantity"
    />
  </View>,
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": undefined,
      }
    }
  />,
  <View
    style={
      Object {
        "flexDirection": "row",
        "minHeight": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fdeb76",
          "flex": 1,
          "marginLeft": 5,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Malt Color"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Country of Origin"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Unit Cost"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Delivery Date"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Amount to Reorder"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Quantity at which to Reorder"
    />
  </View>,
  ");",
]
`;

exports[`MaltForm matches snapshot when values are set 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Current Quantity"
      value="20"
    />
  </View>,
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": undefined,
      }
    }
  />,
  <View
    style={
      Object {
        "flexDirection": "row",
        "minHeight": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fdeb76",
          "flex": 1,
          "marginLeft": 5,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Malt Color"
      value="5"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Country of Origin"
      value="UK"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Unit Cost"
      value="3"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Delivery Date"
      value="01/01/2019"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Amount to Reorder"
      value="30"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Quantity at which to Reorder"
      value="10"
    />
  </View>,
  ");",
]
`;

exports[`YeastForm matches snapshot when no values are set 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Current Quantity"
    />
  </View>,
  <Text
    accessibilityLabel="Yeast Type"
    onPress={[Function]}
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Yeast Type
  </Text>,
  <Text
    accessibilityLabel="Dry"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Dry
  </Text>,
  <RCTSwitch
    accessibilityRole="button"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    value={false}
  />,
  <Text
    accessibilityLabel="Liquid"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Liquid
  </Text>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Unit Cost"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Delivery Date"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Amount to Reorder"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Quantity at which to Reorder"
    />
  </View>,
]
`;

exports[`YeastForm matches snapshot when values are set 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      placeholder="Current Quantity"
      value="2"
    />
  </View>,
  <Text
    accessibilityLabel="Ale"
    onPress={[MockFunction]}
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Ale
  </Text>,
  <Text
    accessibilityLabel="Dry"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Dry
  </Text>,
  <RCTSwitch
    accessibilityRole="button"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    value={true}
  />,
  <Text
    accessibilityLabel="Liquid"
    style={
      Object {
        "color": undefined,
      }
    }
  >
    Liquid
  </Text>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      placeholder="Unit Cost"
      value="10"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      placeholder="Delivery Date"
      value="01/02/2019"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      placeholder="Amount to Reorder"
      value="1"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      placeholder="Quantity at which to Reorder"
      value="1"
    />
  </View>,
]
`;

exports[`otherForm matches snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Current Quantity"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Unit Cost"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Delivery Date"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Amount to Reorder"
    />
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": undefined,
        "borderBottomWidth": 1,
      }
    }
  >
    <TextInput
      autoFocus={false}
      keyboardType="default"
      onBlur={[Function]}
      onFocus={[Function]}
      placeholder="Quantity at which to Reorder"
    />
  </View>,
]
`;
